### ISSUE TITLE
[BUG] VersusEnterAndHoldCriterion misleading for "+/-"-criteria

### ISSUE THREAD

**Issue by nimo23** [ta4j/ta4j#1073]
**Describe the bug**
Some criteria values can have positive and negative values (e.g. PnlCriterion): For such criteria, the `VersusEnterAndHoldCriterion` compares `X` with its `EnterAndHold`, but it does not handle the sign (+/-) in favour of `X`. 

**1. Example: Given:**
myPnl +4%
enterAndHoldPnl: - 4%
=> actual calculation = myPnl / enterAndHoldPnl = 4 / -4 = -1 (it should be +1, because -1 would mean `X `is not better than its `EnterAndHold`)

**2. Example: Given:**
myPnl -4%
enterAndHoldPnl: +4%
=> actual calculation = myPnl / enterAndHoldPnl = -4 / +4 = -1 (correct, because -1 means `X `is not better than its `EnterAndHold`)

**3. Example: Given:**
myPnl -8%
enterAndHoldPnl: -4%
=> actual calculation = myPnl / enterAndHoldPnl = -8 / -4 = 0.5 (ok?)

**Expected behavior**
If a criterion can have both positive and negative values, then `VersusEnterAndHoldCriterion` should handle it in a special way so that the result of the `VersusEnterAndHoldCriterion` can be interpreted unequivocally. 

Maybe we could use another comparision method (instead of `x.dividedBy(enterAndHold)`, we could use "euclidian distance" (but then we have no sign) or any other representative distance calculation, maybe from package `criterion/statistics`).

**Comment by phong-phuong** [comment]
Correct calculation is: (Strategy Account Balance)/(Buy and Hold Account Balance) - 100%.
In your first example, it would look like:
1.04/0.96 - 1.0 = 1.08333333 - 1.0 = 0.08333333, i.e. The strategy when compared to Enter and Hold, outperforms it by 8.33%.

**Comment by phong-phuong** [comment]
Not quite, the formula will depend on what is being compared, and whether "returns" come into the picture.

Consider a portfolio with a start capital of $100.
Strategy A ends with or `returns` $104 back to the user at the end of a backtest.
Strategy B ends with or `returns` $96 back to the user at the end of a backtest.

In this context of returns, the `return` for Strategy A is 104%, and return for Strategy B is 96% which is what ta4j uses.

In Absolute values:
Strategy A Profit: $4
Strategy B Profit: -$4

In percentages:
Strategy A Profit: 4% or 0.04
Strategy B Profit: -4% or -0.04

Absolute difference between Strategy A and Strategy B:
$4 - -$4 = $8
Strategy A outperforms strategy B by $8.

But what if we wanted to compare the profits generated by Strategy A compared to Strategy B?
Well, since we already know Strategy A outperformed by $8, and Strategy return $4, 8/4 = 2, so in this the profits made by Strategy A is 2x Strategy B, in other words, twice as much or 200% of Strategy B.

If we wanted to compare returns generated from strategy A over strategy B as a ratio?
104/96 = 1.08333

If we wanted to compare returns generated from strategy A over strategy B but exclude the initial cost of investment?
104/96 - 1= 0.08333

In the number of winnings example you gave:
Strategy A: 9 wins
Strategy B: 3 wins
Strategy A has 3x more wins than strategy B.

The formula: `(1+9)/(1+3) -1` doesn't come into play, nor make sense because we are not talking about returns or Return on Investment.
We are only doing a direct comparison.

Hope that clears any confusion.

**Comment by phong-phuong** [comment]
No, the formula you are referring to can be used with both percent and absolute values. The formula measures the difference in change compared to a base value.

Going back to the winnings example:
Strat A: 9 wins
Strat B: 3 wins

Applying the formula:
(Ending value - Starting value)/(Starting value)
= (3-9)/9 = -6/9 = -0.667

Or the other version of the formula:
(Ending value)/(Starting value) - 1
= (3/9) - 1 = 0.333 - 1 = -0.667

Which is interpreted as strategy B is 67% worse than strategy A.

**Comment by phong-phuong** [comment]
Or going back to the winnings example where we compare the users strategy vs enter and hold:

User strategy: 9 wins
Enter and hold: 3 wins

Apply the formula:
(User strategy)/(Enter and hold) - 1
= 9/3 - 1 = 3 - 1 = 2

Which is interpreted as the user's strategy when compared to enter and hold is 2x more than the results of Enter and hold.

**Comment by phong-phuong** [comment]
The thing is, sometimes we want to include the base and sometimes we don't

**Comment by phong-phuong** [comment]
That's not the actual formula to be used every time, but I used that + 1 as a substitute of the initial balance because we don't know the starting balance of the user. It is only there to compare the ending capital of the two strategies when we only have the profit or percent to start with.

Compare account balances and directly comparing profits are two separate comparisons.
